metastore/src/java/org/apache/hadoop/hive/metastore/ReplChangeManager.java

```diff
@@ -148,25 +150,37 @@ int recycle(Path path, boolean ifPurge) throws MetaException {
       if (fs.isDirectory(path)) {
         FileStatus[] files = fs.listStatus(path, hiddenFileFilter);
         for (FileStatus file : files) {
-          count += recycle(file.getPath(), ifPurge);
+          count += recycle(file.getPath(), isMove, ifPurge);
         }
       } else {
         Path cmPath = getCMPath(hiveConf, checksumFor(path, fs));
 
-        if (LOG.isDebugEnabled()) {
-          LOG.debug("Moving " + path.toString() + " to " + cmPath.toString());
-        }
-
         // set timestamp before moving to cmroot, so we can
         // avoid race condition CM remove the file before setting
         // timestamp
         long now = System.currentTimeMillis();
         fs.setTimes(path, now, now);
 
-        boolean succ = fs.rename(path, cmPath);
+        boolean success = false;
+        if (isMove) {
+          if (LOG.isDebugEnabled()) {
+            LOG.debug("Moving " + path.toString() + " to " + cmPath.toString());
```

Comment:
I think with slf4j we can use  LOG.debug("Moving {} to {}",path.toString(), cmPath.toString())

this will prevent the creation of string objects for logger which is why the earlier mode of "isDebugEnabled" was checked before calling the log statement, the debug level can be checked within the logger itself.